src/System.Private.CoreLib/shared/System/TimeZoneInfo.cs

```diff
@@ -1989,5 +1979,88 @@ private static void ValidateTimeZoneInfo(string id, TimeSpan baseUtcOffset, Adju
                 }
             }
         }
+
+        private static readonly TimeSpan MaxOffset = TimeSpan.FromHours(14.0);
+        private static readonly TimeSpan MinOffset = -MaxOffset;
+        
+        /// <summary>
+        /// Helper function that validates the TimeSpan is within +/- 14.0 hours
+        /// </summary>
+        internal static bool UtcOffsetOutOfRange(TimeSpan offset) =>
+            offset < MinOffset || offset > MaxOffset;
+
+        private static TimeSpan GetUtcOffset(TimeSpan baseUtcOffset, AdjustmentRule adjustmentRule)
```

Comment:
>GetUtcOffset(Ti [](start = 32, length = 15)

I would to merge this method to the caller one (i.e. IsValidAdjustmentRuleOffest) because it is really small anyway and also we already have many methods with same name.

Reply:
I am seeing you are calling it on 2 places. so it is ok to keep it as it is and not to change it.

---
In reply to: [193195758](https://github.com/dotnet/coreclr/pull/18305#discussion_r193195758) [](ancestors = 193195758)